src/sentry/static/sentry/app/views/settings/organizationMembers/organizationRequestsView.tsx

```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {AccessRequest, Member} from 'app/types';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import {t, tct} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import EmptyMessage from 'app/views/settings/components/emptyMessage';
+import routeTitleGen from 'app/utils/routeTitle';
+import SentryTypes from 'app/sentryTypes';
+
+import InviteRequestRow from './inviteRequestRow';
+import OrganizationAccessRequests from './organizationAccessRequests';
+
+type Props = AsyncView['props'] & {
+  requestList: AccessRequest[];
+  inviteRequests: Member[];
+  updateInviteRequests: (id: string) => void;
+  updateRequestList: (id: string) => void;
+  showInviteRequests: boolean;
+};
+
+type State = AsyncView['state'] & {
+  inviteRequestBusy: Map<string, boolean>;
+};
+
+class OrganizationRequestsView extends AsyncView<Props, State> {
+  static propTypes = {
+    params: PropTypes.object,
+    requestList: PropTypes.array,
+    inviteRequests: PropTypes.array,
+    updateInviteRequests: PropTypes.func,
+    updateRequestList: PropTypes.func,
+    showInviteRequests: PropTypes.bool,
+  };
+
+  static contextTypes = {
+    router: PropTypes.object,
+    organization: SentryTypes.Organization,
+  };
+
+  static defaultProps = {
+    inviteRequests: [],
+  };
+
+  state: State = {
+    inviteRequestBusy: new Map(),
+    ...this.getDefaultState(),
+  };
+
+  getEndpoints() {
+    return [];
+  }
+
+  getTitle() {
+    const {orgId} = this.props.params;
+    return routeTitleGen(t('Requests'), orgId, false);
+  }
+
+  handleApprove = async (id: string, email: string) => {
+    const {
+      params: {orgId},
+      updateInviteRequests,
+    } = this.props;
+
+    this.setState(state => ({
+      inviteRequestBusy: state.inviteRequestBusy.set(id, true),
```

Comment:
This probably isn't great since you're actually directly mutating the state object here. So really what you would want to do is clone the `inviteRequestBusy` map and modify the new one.

If you switch over to a object with the type suggested above you can do `{...state.inviteRequestBusy, [id]: true}`.